### PR TITLE
Fix bug lp:1334657

### PR DESCRIPTION
**Pull request by arthru**
This is the same as https://code.launchpad.net/~arthru/ocb-addons/fix-1334657/+merge/224623

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Hi, @arthru, I can't imagine any situation where the procurement stays in confirmed state, and doesn't make the transition to done, cancel or exception, but if that it's the case, your patch seems right. It's like a process by chunks.

Can you ellaborate that case so that we can also patch that?

Regards.

**Comment by arthru** [comment]
I can't remember what was the data when I encountered that... but I just did... sorry, it was more than a month ago and I did not keep a database with that...

**Comment by pedrobaeza** [comment]
Digging deeper in your patch, I see a potential problem. Imagine you have 600 procurements to check. The current chunk limit is 500. In the first pass, you get full results, changing the state of all of them to other than confirmed (exception, done or cancel). With your patch, next pass will get only 100 results (the 100 that remains in confirmed state), but you have put an offset of 500, so this 100 are never going to be evaluated. Do you agree?

I put :-1:  until we clarify this.

Regards.

File: addons/procurement/schedulers.py [hunk 1]
```diff
@@ -82,7 +82,6 @@ def _procure_confirm(self, cr, uid, ids=None, use_new_cursor=False, context=None
                     if maxdate >= proc.date_planned:
                         wf_service.trg_validate(uid, 'procurement.order', proc.id, 'button_check', cr)
                     else:
-                        offset += 1
                         report_later += 1
```
Comment (hbrunn) [inline addons/procurement/schedulers.py:86]: Shouldn't we check here if state == 'confirmed' and only in this case increase the offset? And I think we have to rebrowse the record to see any changes at all

**Comment by hbrunn** [comment]
closed due to inactivity

**Pull request closed without merging**